src/main/java/org/fungover/thunder/PingHandler.java

```diff
@@ -0,0 +1,31 @@
+package org.fungover.thunder;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.util.logging.Logger;
+
+public class PingHandler {
+    private static final Logger logger = Logger.getLogger(PingHandler.class.getName());
+
+    private PingHandler() {
```

Comment:
Private constructor is fine by me, no need to throw exception :)

Reply:
@BarbieQkiller Even if the constructor is private it's actually possible to use it with reflection. And if we look at Hibernate for example that can create instances of Entity classes or ValueObjects it doesn't care about the access modifier. 
https://stackoverflow.com/questions/71824535/how-to-make-sure-instance-of-a-class-with-a-private-constructor-can-not-be-creat

That's why it can be a good practice in a bigger enterprise application to throw an exception from the private constructor in a utility class to prevent this happening without noticing.